src/internal/git/push.go

```diff
@@ -18,12 +18,38 @@ const offlineRemoteName = "offline-downstream"
 const onlineRemoteRefPrefix = "refs/remotes/" + onlineRemoteName + "/"
 
 func PushAllDirectories(localPath string) error {
-	// Establish a git tunnel to send the repos
-	tunnel := k8s.NewZarfTunnel()
-	tunnel.Connect(k8s.ZarfGit, false)
-	defer tunnel.Close()
+	gitServerInfo := config.GetGitServerInfo()
+	gitServerURL := ""
+
+	if gitServerInfo.InternalServer {
+		// This is an internal gitea server so we'll use what we normally do..
+		// Establish a git tunnel to the internal gitea pod to send the repos
+		tunnel := k8s.NewZarfTunnel()
+		tunnel.Connect(k8s.ZarfGit, false)
+		defer tunnel.Close()
+
+		gitServerURL = fmt.Sprintf("http://%s", tunnel.Endpoint())
+	} else {
+		// This is an external git server so we'll use whatever we were provided with
+		gitServerURL = gitServerInfo.GitAddress
+
+		// Add a port to the URL if it was provided
+		if gitServerInfo.GitPort != 0 {
+			gitServerURL = fmt.Sprintf("%s:%d", gitServerInfo.GitAddress, gitServerInfo.GitPort)
+		}
 
-	tunnelUrl := fmt.Sprintf("http://%s", tunnel.Endpoint())
+		// If this URL points to a resource within the cluster, create a tunnel to it
+		if strings.Contains(gitServerURL, "svc.cluster.local") {
```

Comment:
This is probably fine since this is super edge-casey, but my brain immediately goes to "but what about https://svc.cluster.local.example.com!"

Reply:
That's a good edge-case! I changed the way we check the URL string to avoid situations like this.

Reply:
:tada: